azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -381,7 +381,11 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
       // inject cluster jars and native libraries into jobs through properties
       // User's job props should take precedence over cluster props
       Props finalProps = getClusterSpecificJobProps(targetCluster, jobProps, pluginLoadProps);
+      Props nonOverriddableClusterProps = getClusterSpecificNonOverridableJobProps(finalProps);
       finalProps.putAll(jobProps);
+      // add router-specific properties that are critical for job execution as the last step
```

Comment:
Please add cautionary comment, perhaps something in all caps to make sure no one adds props after that code.

Reply:
addressed